src/lib/services/LocalCatalog.ts

```diff
@@ -52,24 +52,23 @@ export default class LocalCatalog implements RuleCatalog {
 		}
 		// If we actually do have filters, we'll want to iterate over all of them and see which ones
 		// correspond to a path in the catalog.
-		// Since categories and rulesets are both just NamedPaths, we can put both types of
+		// Since categories and rulesets are both just RuleGroups, we can put both types of
 		// path into a single array and return that.
-		const foundPaths: RuleGroup[] = [];
+		const foundRuleGroups: RuleGroup[] = [];
 		for (const filter of filters) {
 			// For now, we only care about filters that act on rulesets and categories.
-			const type = filter.filterType;
-			if (type === FilterType.CATEGORY || type === FilterType.RULESET) {
+			if (isRuleGroupFilter(filter)) {
```

Comment:
Does this now fix the Engine filter on PMD issue? Previously, the bug was that the Engine filter would not be handled here at all and an empty RuleGroup array was returned. Looks like it should be gone now.

Reply:
Unfortunately, this doesn't fix the issue.